gateserver/utils/structparse.py

```diff
@@ -0,0 +1,47 @@
+from collections import namedtuple
+from struct import Struct
+from enum import Enum
+from . import unzip
+
+ENDIANITY = '<'  # little-endian, no alignment (i.e. packed)
+
+class t:
+    """pieces of struct format strings: docs.python.org/3/library/struct.html"""
+    uint8 = 'B'
+    bytes = lambda sz: '{}s'.format(sz)
+
+class MyStructMixin:
```

Comment:
Tail by sa mal volat v kazdej strukture inak, nie vzdy "tail". Napriklad v requeste by mal byt `request.data`. (Napriklad by sa to mohlo urcovat ako `mystruct(..., tail_name='request')`.) A mal by tiez byt clenom tuplu.
Kazda struktura by mala mat iba unpack, nie aj unpack aj unpack_with_tail - nech to nie je zodpovednost pouzivatela struktury.
Struktury, co nemaju ziaden tail, by mali pri unpacknuti nadbytocnych dat rovno hadzat exception.